canonical/traefik-k8s-operator
# Implement V1 of certificate_transfer

## Issue
The new version (V1) of the certificate_transfer_interface is incompatible of its V0 version. Applications that implement V1 of cert transfer won't be able to share their CAs with Traefik.

## Solution

This change is to allow application the implement the V1 version of the interface to integrate with Traefik to provide their CAs that Traefik needs to trust.
**Note**: The solution might seem a bit hacky, I tried to follow the same solution with `IngressPerAppV0` and `IngressPerAppV1`.

## Context



## Testing Instructions
- Deploy Traefik from this branch
- Deploy `notary-k8s` from [this](https://github.com/canonical/notary-k8s-operator/pull/31) branch
- Deploy self-signed-certificates
- Integrate Notary and self-signed-certificates using the `certificates` interface
- Integrate Notary and Traefik using the `receive-ca-cert-v1` interface
- Observe the CA file being created in the correct directory in the Traefik container
- Remove the integration and observe the file being removed


## Upgrade Notes
